apps/wallet/src/ui/app/hooks/useSortedCoinsByCategories.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { usePinnedCoinTypes } from '_app/hooks/usePinnedCoinTypes';
+import { useRecognizedPackages } from '_app/hooks/useRecognizedPackages';
+import { type CoinBalance as CoinBalanceType } from '@mysten/sui.js/client';
+import { useMemo } from 'react';
+
+export function useSortedCoinsByCategories(coinBalances: CoinBalanceType[]) {
```

Comment:
migrated from `TokenDetails.tsx`. This is used in a different spot for DEFI feature

Reply:
Just calling out that this approach to sorting really doesn't work when this list is paginated. I know it's not your code here so not a blocking issue, just something we do actually need to sort out eventually.